### ISSUE TITLE
Simple source detection function for FluxMaps 

### ISSUE THREAD

**Issue by registerrier** [gammapy/gammapy#4470]
**Is your feature request related to a problem? Please describe.**
The current find_peaks algorithm offers a minimalist set of data from a significance map. Much more can be extracted from `FluxMaps`: position in radec, and associated error, flux end flux error (in various sed type), excess counts. 
  
**Describe the solution you'd like**
A first approach could be to create a function in `gammapy.estimators.utils` to compute this.

**Additional context**
Note we have to understand whether this type of algorithm really belongs to the `Estimator` abstraction (DL4 to DL5). It is rather a tool extracting information from an estimate and building some higher level information. I am not sure this is really DL6 data, though, as this would rather be a list of source candidates without spectral information.

Similarly, the data format for such a product is not fully clear. At first a regular `Table` is fine. It could be partly converted to a `Models` object to allow for easy fitting.

**Comment by adonath** [comment]
I agree this would be very useful. Here are just some links to existing functionality in `photutils`:

- https://photutils.readthedocs.io/en/stable/detection.html
- https://photutils.readthedocs.io/en/stable/segmentation.html
- https://photutils.readthedocs.io/en/stable/morphology.html
- https://photutils.readthedocs.io/en/stable/aperture.html

With a combination of the linked functionality above one could support extended sources as well or perform a simple deblending, segmentation and estimation of morphological properties.

**Comment by Astro-Kirsty** [comment]
I have added a PR to address the initial request. 
To extract the sub-pixel location and therefore further features from the maps is more difficult. The method I have attempted is to utilise the astropy `modeling.functional_models.Gaussian2D `to fit the peaks (found through `find_peaks_in_flux_map`). In general, this works well when you have peaks which are separated by some distance. It would be good to discuss this further, and if anyone has suggestions to let me know.
